### PR TITLE
Optimize code logic to avoid some unexcepted exception

### PR DESCRIPTION
**Pull request by hpejustin**
1.  Handle some error has been ignored
2. Change some hard code value to variables
3. Make log info more appropriate

### CODE REVIEW

File: cmd/poseidon/poseidon.go [hunk 1]
```diff
@@ -31,6 +31,11 @@ import (
 	"github.com/golang/glog"
 )
 
+const (
+	Interval = 2 * time.Second
+	TimeOut  = 10 * time.Minute
```
Comment (m1093782566) [inline cmd/poseidon/poseidon.go]: nit: Timeout

And what interval? what timeout?

File: cmd/poseidon/poseidon.go [hunk 2]
```diff
@@ -95,7 +103,7 @@ func WaitForFirmamentService(fc firmament.FirmamentSchedulerClient) {
 }
 
 func main() {
-	glog.Info("Starting Poseidon...", config.GetFirmamentAddress())
+	glog.Info("Starting Poseidon...")
```
Comment (m1093782566) [inline cmd/poseidon/poseidon.go]: Please add the address

Comment (hpejustin) [reply cmd/poseidon/poseidon.go]: OK, updated.

File: cmd/poseidon/poseidon.go (hunk 1 above)
Comment (hpejustin) [reply cmd/poseidon/poseidon.go]: OK, I have updated.

**Comment by m1093782566** [comment]
/cc @hanxiaoshuai 

for review.

File: cmd/poseidon/poseidon.go [hunk 3]
```diff
@@ -82,8 +87,11 @@ func schedule(fc firmament.FirmamentSchedulerClient) {
 func WaitForFirmamentService(fc firmament.FirmamentSchedulerClient) {
 	// TODO(jiaxuanzhou): Need to metric the wait latency of firmament service?
 	serviceReq := new(firmament.HealthCheckRequest)
-	err := wait.PollImmediate(2*time.Second, 10*time.Minute, func() (bool, error) {
-		ok, _ := firmament.Check(fc, serviceReq)
+	err := wait.PollImmediate(FirmamentHealthCheckInterval, FirmamentHealthCheckTimeOut, func() (bool, error) {
+		ok, err := firmament.Check(fc, serviceReq)
+		if err != nil {
+			panic(err)
```
Comment (jiaxuanzhou) [inline cmd/poseidon/poseidon.go]: we need to  throw panic here ?

File: cmd/poseidon/poseidon.go [hunk 4]
```diff
@@ -31,6 +31,11 @@ import (
 	"github.com/golang/glog"
 )
 
+const (
+	FirmamentHealthCheckInterval = 2 * time.Second
+	FirmamentHealthCheckTimeOut  = 10 * time.Minute
```
Comment (hanxiaoshuai) [inline cmd/poseidon/poseidon.go]: nit: FirmamentHealthCheckTimeOut -> FirmamentHealthCheckTimeout

File: cmd/poseidon/poseidon.go [hunk 5]
```diff
@@ -95,7 +103,8 @@ func WaitForFirmamentService(fc firmament.FirmamentSchedulerClient) {
 }
 
 func main() {
-	glog.Info("Starting Poseidon...", config.GetFirmamentAddress())
+
+	glog.Info("Starting Poseidon with firmament address %s.", config.GetFirmamentAddress())
```
Comment (hanxiaoshuai) [inline cmd/poseidon/poseidon.go]: nit: glog.Info --> glog.Infof

Comment (hpejustin) [reply cmd/poseidon/poseidon.go]: Okay, updated.

File: cmd/poseidon/poseidon.go (hunk 4 above)
Comment (hpejustin) [reply cmd/poseidon/poseidon.go]: Okay,updated.

**Pull request merged**